fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java

```diff
@@ -121,14 +127,17 @@ public long beginTransaction(long dbId, String label, String coordinator, LoadJo
             if (txnLabels != null && txnLabels.containsKey(label)) {
                 throw new LabelAlreadyExistsException("label already exists, label=" + label);
             }
-            if (runningTxnNums.get(dbId) != null 
+            if (runningTxnNums.get(dbId) != null
                     && runningTxnNums.get(dbId) > Config.max_running_txn_num_per_db) {
-                throw new BeginTransactionException("current running txns on db " + dbId + " is " 
-                    + runningTxnNums.get(dbId) + ", larger than limit " + Config.max_running_txn_num_per_db);
+                throw new BeginTransactionException("current running txns on db " + dbId + " is "
+                        + runningTxnNums.get(dbId) + ", larger than limit " + Config.max_running_txn_num_per_db);
             }
             long tid = idGenerator.getNextTransactionId();
             LOG.debug("beginTransaction: tid {} with label {} from coordinator {}", tid, label, coordinator);
             TransactionState transactionState = new TransactionState(dbId, tid, label, sourceType, coordinator);
+            if (transactionCallback != null) {
```

Comment:
I think you can add callback to `TransactionState` constructor to avoid this `if` statement. And I prefer `TransactionStateChangeListener` to `Callback`

Reply:
I will change it.